engine/access/ingestion/engine.go

```diff
@@ -376,6 +376,17 @@ func (e *Engine) processFinalizedBlock(blockID flow.Identifier) error {
 		}
 	}
 
+	// skip requesting collections, if this block is below the last full block height
+	// this means that either we have already received these collections, or the block
+	// may contain unverifiable guarantees (in case this node has just joined the network)
+	lastFullBlockHeight, err := e.blocks.GetLastFullBlockHeight()
+	if err != nil {
+		return fmt.Errorf("could not get last full block height: %w", err)
+	}
+	if block.Header.Height <= lastFullBlockHeight {
+		return nil
```

Comment:
Should we add a log message here, something like "skipping requesting collections for block height < last full block height"?